components/list/list-item-mixin.js

```diff
@@ -659,6 +661,9 @@ export const ListItemMixin = superclass => class extends composeMixins(
 			'd2l-list-item-color-inner': true,
 			'd2l-skeletize': this.color
 		};
+		const bottomBorderClasses = {
+			'hide-bottom-border': this._showAddButton && (!this._hasNestedList || this._hasNestedListAddButton)
```

Comment:
I added a class for this rather than exposing another property. I can switch to that if desired since it would be more consistent with this particular code.